src/axom/sidre/core/Group.hpp

```diff
@@ -1179,14 +1179,39 @@ class Group
   /*!
    * \brief Load the Group from a file.
    *
-   * \param path      file path
-   * \param protocol  I/O protocol
+   * \param path     file path
+   * \param new_name the group name stored in the file
+   * \param protocol I/O protocol
    * \param preserve_contents   Preserve existing contents of group if true
    */
   void load(const std::string& path,
+            std::string & new_name,
             const std::string& protocol = SIDRE_DEFAULT_PROTOCOL,
             bool preserve_contents = false);
 
```

Comment:
Relatedly, this API change carries some danger with its addition of a required string argument right before an optional string argument.  If a calling code currently has a line

`group.load("filename.hdf5", "sidre_hdf5");`

It will still compile once these changes are installed, but now "sidre_hdf5" will be treated as a group name rather than as a protocol.

Also, there are a couple of calls in IOManager.cpp (lines 315 and 791) where I'm not sure how they compile with these changes to load().

Reply:
Good call @nselliott 

One possible way around this would be to use the ``new_name`` parameter only in an overload with no optional parameters, and to put it at the end.
E.g. the two signatures would be
```cpp
bool load( const string& file, 
           const string& protocol = SIDRE_DEFAULT_PROTOCOL, 
           bool preserve_contents = false);

bool load( const string& file, 
           const string& protocol,  // note: not optional
           bool preserve_contents,  // note: not optional
           string& new_name);
```

Reply:
Good point, @nselliott .  Thanks, @kennyweiss , I like that idea.

Reply:
done